pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -296,26 +296,33 @@ func inferenceServiceStatusEqual(s1, s2 v1beta1api.InferenceServiceStatus, deplo
 }
 
 func (r *InferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager, deployConfig *v1beta1api.DeployConfig, ingressConfig *v1beta1api.IngressConfig) error {
-	if deployConfig.DefaultDeploymentMode == string(constants.RawDeployment) {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1beta1api.InferenceService{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
-	} else if ingressConfig.DisableIstioVirtualHost == false {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1beta1api.InferenceService{}).
-			Owns(&knservingv1.Service{}).
-			Owns(&istioclientv1beta1.VirtualService{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
+	ksvcFound, err := utils.IsCrdAvailable(mgr.GetConfig(), "serving.knative.dev/v1", "Service")
+	if err != nil {
+		return err
+	}
+
+	vsFound, err := utils.IsCrdAvailable(mgr.GetConfig(), "networking.istio.io/v1beta1", "VirtualService")
```

Comment:
Move these strings to constants and consider using a loop to check all of these CRDs since we may need to check them for different versions, etc.

Reply:
I do agree on moving to constants, I'll do it.

About a loop for checking different versions, I'm not sure that's useful. Read more in [my reply to Jooho](https://github.com/kserve/kserve/pull/3472/files#r1503095746).

Reply:
Guess the best way is to have a const (array in case of more versions) and check those.

Reply:
I modified to using constants and 3rd-party library-provided variables